core/shared/src/main/scala/org/http4s/ServerSentEvent.scala

```diff
@@ -97,40 +91,31 @@ object ServerSentEvent {
         Pull.output1(sse) >> go(emptyBuffer, None, None, None, emptyBuffer, stream)
       }
 
-      def handleLine(
-          field: String,
-          value: String,
-          stream: Stream[F, String],
-      ): Pull[F, ServerSentEvent, Unit] =
-        field match {
-          case "" =>
-            go(dataBuffer, eventType, id, retry, commentBuffer.append(value), stream)
-          case "event" =>
-            go(dataBuffer, Some(value), id, retry, commentBuffer, stream)
-          case "data" =>
-            go(dataBuffer.append(value), eventType, id, retry, commentBuffer, stream)
-          case "id" =>
-            val newId = EventId(value)
-            go(dataBuffer, eventType, Some(newId), retry, commentBuffer, stream)
-          case "retry" =>
-            val newRetry = Try(value.toLong).toOption.orElse(retry)
-            go(dataBuffer, eventType, id, newRetry, commentBuffer, stream)
-          case _ =>
-            go(dataBuffer, eventType, id, retry, commentBuffer, stream)
-        }
-
       stream.pull.uncons1.flatMap {
         case None =>
           Pull.done
         case Some(("", stream)) =>
           dispatch(stream)
         case Some((s, stream)) =>
-          (FieldSeparator.split(s, 2): @unchecked) match {
-            case Array(field, value) =>
-              handleLine(field, value, stream)
-            case Array(line) =>
-              handleLine(line, "", stream)
-          }
+          if (s.startsWith("data:")) {
+            go(dataBuffer.append(dropPrefix(s, 5)), eventType, id, retry, commentBuffer, stream)
+          } else if (s.startsWith("id:")) {
+            go(dataBuffer, eventType, Some(EventId(dropPrefix(s, 3))), retry, commentBuffer, stream)
+          } else if (s.startsWith("event:")) {
+            go(dataBuffer, Some(dropPrefix(s, 6)), id, retry, commentBuffer, stream)
+          } else if (s.startsWith("retry:")) {
+            val newRetry = Try(dropPrefix(s, 6).toLong).toOption.orElse(retry)
+            go(dataBuffer, eventType, id, newRetry, commentBuffer, stream)
+          } else if (s.startsWith(":")) {
+            go(dataBuffer, eventType, id, retry, commentBuffer.append(dropPrefix(s, 1)), stream)
+          } else
+            s match {
+              case "data" => go(dataBuffer.append(""), eventType, id, retry, commentBuffer, stream)
+              case "id" =>
+                go(dataBuffer, eventType, Some(EventId("")), retry, commentBuffer, stream)
+              case "event" => go(dataBuffer, Some(""), id, retry, commentBuffer, stream)
```

Comment:
Haven't we missed the `retry` here from https://github.com/http4s/http4s/pull/7528/files?diff=split&w=1#diff-23df5df5d5a43a571f052ee04603672553424b84c20191b383127bca4b0d7255L115-L117?

Reply:
hi, it should be ignored according to the spec, same as the next case:

taking about no colon:
```
Otherwise, the string is not empty but does not contain a U+003A COLON character (:)

    [Process the field](https://www.w3.org/TR/2012/WD-eventsource-20120426/#processField) using the steps described below, using the whole line as the field name, and the empty string as the field value.
```

and

```
If the field name is "retry"

    If the field value consists of only characters in the range U+0030 DIGIT ZERO (0) to U+0039 DIGIT NINE (9), then interpret the field value as an integer in base ten, and set the event stream's [reconnection time](https://www.w3.org/TR/2012/WD-eventsource-20120426/#concept-event-stream-reconnection-time) to that integer. Otherwise, ignore the field.```

Reply:
https://www.w3.org/TR/2012/WD-eventsource-20120426/